### PR TITLE
Refactor auth config

### PR DESCRIPTION
**Pull request by msteffen**
Pull a lot of auth config code into `src/server/auth/server/config.go`. Also, change the internal representation of the auth config. Full design is at  https://github.com/pachyderm/pachyderm/issues/3949, but two highlights:
- no auth config in etcd => "default" auth config is the active config
- once an auth config is in etcd, there is no way to delete it. Sending an empty `SetConfiguration` RPC writes the default auth config into etcd (with the latest version)

(fix https://github.com/pachyderm/pachyderm/issues/3949)

### CODE REVIEW

File: src/server/auth/server/api_server.go [hunk 1]
```diff
@@ -74,6 +73,8 @@ const (
 	SamlPort = 654
 )
 
+var defaultAuthConfig = authclient.AuthConfig{}
```
Comment (kevindelgado) [inline src/server/auth/server/api_server.go]: It is here where I want to modify the default auth config to be 
```{
  "live_config_version": 1, // prevents RMW errors
  "id_providers": [
    {
      "name": "GitHub",
      "description": "oauth-based authentication with github.com",
      "github": {}
    }
  ]
}
``` 
right?

File: src/server/auth/server/api_server.go [hunk 2]
```diff
@@ -2258,16 +2206,16 @@ func (a *apiServer) canonicalizeSubject(ctx context.Context, subject string) (st
...
 
 	// check prefix against config cache
-	if a.configCache != nil {
-		if prefix == a.configCache.IDP.Name {
-			return subject, nil
-		}
-		if prefix == path.Join("group", a.configCache.IDP.Name) {
-			return subject, nil // TODO(msteffen): check if this IdP supports groups
+	if config := a.getCacheConfig(); config != nil {
+		for _, idp := range config.IDPs {
```
Comment (kevindelgado) [inline src/server/auth/server/api_server.go]: From our discussion, we are going to continue to canonicalize the subject based only on the IDP's name and not on anything that is in the various *Options fields, correct?

File: src/server/auth/server/api_server.go (hunk 1 above)
Comment (msteffen) [reply src/server/auth/server/api_server.go]: yup, exactly

File: src/server/auth/server/api_server.go (hunk 2 above)
Comment (msteffen) [reply src/server/auth/server/api_server.go]: I'm not 100% sure I undrestand the question, but I think that's right (this is also what we currently, do, if I understand the question).

To be clear, and maybe a little pendatic
1) I think the plan for now is that post-change, an IDP's `name` is `github` if and only if it has a `GitHubOptions`. It's just a matter of whether there's an IDP with `GitHubOptions`), or equivalently, an IDP whose `name` is `github`.
2) a SAML IDP can give you two subject types if it supports groups:
- `<idp.name>:<subject>` (e.g. `okta:matthew`)
- `group/<idp.name>:<subject>` (e.g. `group/okta:engineering`).
but we don't currently check the IDP to make sure `saml.group_attribute` is set before allowing `group/<idp_name>`-type subjects through `canonicalizeSubject`. Perhaps we should but I think it's fine to keep that outside the scope of this change.

**Pull request merged**